docs/reference/search/async-search.asciidoc

```diff
@@ -0,0 +1,200 @@
+--------------------------------------------------
+// TEST[continued s/FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=/\${body.id}/]
+
+[source,console-result]
+--------------------------------------------------
+{
+  "id" : "FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=",
+  "version" : 2, <1>
+  "is_partial" : true,
+  "is_running" : true,
+  "start_time_in_millis" : 1583945890986,
+  "expiration_time_in_millis" : 1584377890986,
+  "response" : {
+    "took" : 12144,
+    "timed_out" : false,
+    "num_reduce_phases" : 38,
+    "_shards" : {
+      "total" : 562,
+      "successful" : 188,
+      "skipped" : 0,
+      "failed" : 0
+    },
+    "hits" : {
+      "total" : {
+        "value" : 456433,
+        "relation" : "eq"
+      },
+      "max_score" : null,
+      "hits" : [ ]
+    },
+    "aggregations" : { <2>
+      "sale_date" :  {
+        "buckets" : []
+      }
+    }
+  }
+}
+--------------------------------------------------
+// TESTRESPONSE[s/FmRldE8zREVEUzA2ZVpUeGs2ejJFUFEaMkZ5QTVrSTZSaVN3WlNFVmtlWHJsdzoxMDc=/$body.id/]
+// TESTRESPONSE[s/"is_partial" : true/"is_partial" : false/]
+// TESTRESPONSE[s/"is_running" : true/"is_running" : false/]
+// TESTRESPONSE[s/1583945890986/$body.start_time_in_millis/]
+// TESTRESPONSE[s/1584377890986/$body.expiration_time_in_millis/]
+// TESTRESPONSE[s/"took" : 12144/"took": $body.response.took/]
+// TESTRESPONSE[s/"total" : 562/"total": $body.response._shards.total/]
+// TESTRESPONSE[s/"successful" : 188/"successful": $body.response._shards.successful/]
+// TESTRESPONSE[s/"value" : 456433/"value": $body.response.hits.total.value/]
+// TESTRESPONSE[s/"buckets" : \[\]/"buckets": $body.response.aggregations.sale_date.buckets/]
+// TESTRESPONSE[s/"num_reduce_phases" : 38,//]
+
+<1> The returned `version` is useful to identify whether the response contains
+additional results compared to previously obtained responses. If the version
+stays the same, no new results have become available, otherwise a higher version
+number indicates that more shards have completed their execution of the query
+and their partial results are also included in the response.
+<2> Partial aggregations results, coming from the shards that have already
+completed the execution of the query.
+
+NOTE: When results are sorted by a numeric field, shards get sorted based on
+minimum and maximum value that they hold for that field, hence partial
+results become available following the sort criteria that was requested.
+
+The `wait_for_completion` parameter, which defaults to `1`, can also be provided
+when calling the Get Async Search API, in order to wait for the search to be
+completed up until the provided timeout. Final results will be returned if
+available before the timeout expires, otherwise the currently available results
+will be returned once the timeout expires.
+
+The `keep_alive` parameter, which defaults to `5d` (five days), specifies
```

Comment:
The `keep_alive` for the get api is a bit different since it allows to extend/change the expiration time for an existing id. Maybe good to make the distinction here since the default for get is to keep the current expiration time.